src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -725,7 +747,13 @@ private static Expression BindParameterFromBody(ParameterInfo parameter, bool al
         {
             if (factoryContext.JsonRequestBodyType is not null)
             {
-                throw new InvalidOperationException("Action cannot have more than one FromBody attribute.");
+                factoryContext.HasAnotherBodyParameter = true;
+                var parameterName = parameter.Name;
+                if (parameterName is not null && factoryContext.TrackedParameters.ContainsKey(parameterName))
+                {
+                    factoryContext.TrackedParameters.Remove(parameterName);
+                    factoryContext.TrackedParameters.Add(parameter.Name!, "We failed to infer this parameter. Did you forget to inject it as a Service in the DI container?");
```

Comment:
```suggestion
                    factoryContext.TrackedParameters.Add(parameter.Name!, "UNKNOWN");
```

I think the following is a bit hard to read:

```
  Unhandled exception. System.InvalidOperationException: Failure to infer one or more parameters. Below is the list of parameters we found:

Parameter          Source

sampleDTO  Body Attribute
userRepo We failed to infer this parameter. Did you forget to inject it as a Service?

   at Microsoft.AspNetCore.Http.RequestDelegateFactory.CreateArguments(ParameterInfo[] parameters, FactoryContext factoryContext) in C:\Users\rassumani\bench\aspnetcore\src\Http\Http.Extensions\src\RequestDelegateFactory.cs:line 209
   at Microsoft.AspNetCore.Http.RequestDelegateFactory.CreateTargetableRequestDelegate(MethodInfo methodInfo, RequestDelegateFactoryOptions options, Expression targetExpression) in C:\Users\rassumani\bench\aspnetcore\src\Http\Http.Extensions\src\RequestDelegateFactory.cs:line 170
   ...
```

This would be a bit better:

```
  Unhandled exception. System.InvalidOperationException: Failure to infer one or more parameters. Below is the list of parameters we found:

Parameter |            Source
----------+------------------
sampleDTO |  Body (Attribute)
userRepo  |            UNKOWN


Did you forget to inject the "UNKNOWN" parameters as a Service?


   at Microsoft.AspNetCore.Http.RequestDelegateFactory.CreateArguments(ParameterInfo[] parameters, FactoryContext factoryContext) in C:\Users\rassumani\bench\aspnetcore\src\Http\Http.Extensions\src\RequestDelegateFactory.cs:line 209
   at Microsoft.AspNetCore.Http.RequestDelegateFactory.CreateTargetableRequestDelegate(MethodInfo methodInfo, RequestDelegateFactoryOptions options, Expression targetExpression) in C:\Users\rassumani\bench\aspnetcore\src\Http\Http.Extensions\src\RequestDelegateFactory.cs:line 170
   ...
```

Adding the `|` and `-` would require more than just the suggested fix of course but I think it will help. Can you show what this looks like on the developer exception page?

Reply:
Also, do we get the same "Failure to infer one or more parameters." error when all the parameters have explicit attributes? If there actually are multiple `[FromBody]` attributes we should probably just continue saying that's not allowed instead of saying we couldn't infer the source.

Reply:
Do we have a logger formatter or console formatter in aspnetcore that I can use ?

Reply:
I am hoping an analyzer will prevent the double explicit  ```[FromBody]``` right away.

Reply:
New formatting. Not perfect, visually readable.
``` 
Unhandled exception. System.InvalidOperationException: Failure to infer one or more parameters. 
Below is the list of parameters we found:

Parameter           |Source
------------------------------------------
id                  | Route (Inferred)
dto                 | Body (Attribute)
dto1                | We failed to infer this parameter. Did you forget to inject it as a Service in the DI container?
repo                | Service (Attribute)
repo1               | We failed to infer this parameter. Did you forget to inject it as a Service in the DI container?

```

Reply:
"Did you mean to register the "UNKNOWN" parameters as a Service?"

Reply:
```
Unhandled exception. System.InvalidOperationException: Failure to infer one or more parameters.
Below is the list of parameters that we found:

Parameter           |Source
---------------------------------------------------------------------------------
id                  | Route (Inferred)
dto                 | Body (Attribute)
dto1                | UNKNOWN
repo                | Service (Attribute)
repo1               | UNKNOWN


Did you mean to register the "UNKNOWN" parameters as a Service?


   at Microsoft.AspNetCore.Http.RequestDelegateFactory.CreateArguments(ParameterInfo[] parameters, FactoryContext factoryContext) in C:\Users\rassumani\bench\aspnetcore\src\Http\Http.Extensions\src\RequestDelegateFactory.cs:line 196
   at Microsoft.AspNetCore.Http.RequestDelegateFactory.CreateTargetableRequestDelegate(MethodInfo methodInfo, RequestDelegateFactoryOptions options, Expression targetExpression) in C:\Users\rassumani\bench\aspnetcore\src\Http\Http.Extensions\src\RequestDelegateFactory.cs:line 165
   at Microsoft.AspNetCore.Http.RequestDelegateFactory.Create(Delegate action, RequestDelegateFactoryOptions options) in C:\Users\rassumani\bench\aspnetcore\src\Http\Http.Extensions\src\RequestDelegateFactory.cs:line 84


```